controllers/components/datasciencepipelines/datasciencepipelines_support.go

```diff
@@ -42,20 +49,38 @@ var (
 		cluster.OpenDataHub:      "overlays/odh",
 		cluster.Unknown:          "overlays/odh",
 	}
+
+	paramsPath = path.Join(odhdeploy.DefaultManifestPath, ComponentName, "base")
 )
 
-func paramsPath() types.ManifestInfo {
+func manifestPath(p common.Platform) types.ManifestInfo {
 	return types.ManifestInfo{
 		Path:       odhdeploy.DefaultManifestPath,
 		ContextDir: ComponentName,
-		SourcePath: "base",
+		SourcePath: overlaysSourcePaths[p],
 	}
 }
 
-func manifestPath(p common.Platform) types.ManifestInfo {
-	return types.ManifestInfo{
-		Path:       odhdeploy.DefaultManifestPath,
-		ContextDir: ComponentName,
-		SourcePath: overlaysSourcePaths[p],
+func computeParamsMap(rr *types.ReconciliationRequest) (map[string]string, error) {
+	dsp, ok := rr.Instance.(*componentApi.DataSciencePipelines)
+	if !ok {
+		return nil, fmt.Errorf("resource instance %v is not a componentApi.DataSciencePipelines", rr.Instance)
+	}
+
+	data, err := json.Marshal(dsp.Spec.PreloadedPipelines)
+	if err != nil {
+		return nil, fmt.Errorf("marshalling preloaded pipelines failed: %w", err)
+	}
+
+	data, err = json.Marshal(string(data))
+	if err != nil {
+		return nil, fmt.Errorf("marshalling preloaded pipelines failed: %w", err)
+	}
+
+	extraParamsMap := map[string]string{
+		managedPipelineParamsKey: string(data),
+		platformVersionParamsKey: rr.Release.Version.String(),
```

Comment:
why not have the platform version set done in `func (s *componentHandler) Init(_ common.Platform)`

Reply:
done